controllers/virtualmachinepublishrequest/virtualmachinepublishrequest_controller.go

```diff
@@ -663,23 +701,61 @@ func (r *Reconciler) getPublishRequestTask(ctx *pkgctx.VirtualMachinePublishRequ
 	}
 
 	if len(tasks) == 0 {
-		logger.V(5).Info("task doesn't exist", "actID", actID, "descriptionID", TaskDescriptionID)
+		logger.V(5).Info("task doesn't exist", "actID", actID, "descriptionID", OVFCaptureTaskDescriptionID)
 		return nil, nil
 	}
 
 	// Return the first task.
 	// We would never send multiple CreateOvf requests with the same actID,
 	// so that we should never get multiple tasks.
 	publishTask := &tasks[0]
-	if publishTask.DescriptionId != TaskDescriptionID {
-		err = fmt.Errorf("failed to find expected task %s, found %s instead", TaskDescriptionID, publishTask.DescriptionId)
+	if publishTask.DescriptionId != OVFCaptureTaskDescriptionID {
+		err = fmt.Errorf("failed to find expected task %s, found %s instead", OVFCaptureTaskDescriptionID, publishTask.DescriptionId)
 		logger.Error(err, "task doesn't exist")
 		return nil, err
 	}
 
 	return publishTask, nil
 }
 
+// getCloneVMTask gets task with description id VirtualMachine.clone and whose result.Value is the MOID of the target
+// template.
+func (r *Reconciler) getCloneVMTask(ctx *pkgctx.VirtualMachinePublishRequestContext) (*vimtypes.TaskInfo, error) {
+	vmPubReq := ctx.VMPublishRequest
+	targetItemName := vmPubReq.Status.TargetRef.Item.Name
+
+	vm := &vmopv1.VirtualMachine{}
+	objKey := client.ObjectKey{Name: vmPubReq.Status.SourceRef.Name, Namespace: vmPubReq.Namespace}
+	err := r.Get(ctx, objKey, vm)
+	if err != nil {
+		ctx.Logger.Error(err, "failed to get VirtualMachine", "vm", objKey)
+		return nil, err
+	}
```

Comment:
No need to log the error if it's wrapped into the returned error:
```suggestion
	if err := r.Get(ctx, objKey, vm); err != nil {
		return nil, fmt.Errorf("failed to get VirtualMachine %q: %w", objKey, err)
	}
```